src/string_func.h

```diff
@@ -160,6 +162,13 @@ inline const char *Utf8PrevChar(const char *s)
 	return ret;
 }
 
+inline std::string::iterator Utf8PrevChar(std::string::iterator &s)
+{
+	auto *start = &*s;
+	auto *end = Utf8PrevChar(start);
+	return s - (start - end);
```

Comment:
```suggestion
	auto *cur = &*s;
	auto *prev = Utf8PrevChar(cur);
	return s - (cur - prev);
```

The `end` threw me for a bit :P

Reply:
Yeah, `prev` makes more sense than `end` here to me.